src/reducers/computeRenderedColumns.js

```diff
@@ -138,6 +138,87 @@ function calculateRenderedColumnRange(state, columnAnchor) {
   };
 }
 
+/**
+ * Determine the range amd offsets of column groups to be rendered (buffer and viewport)
+ *
+ * NOTE (jordan) This alters state so it shouldn't be called
+ * without state having been cloned first.
+ *
+ * @param {!Object} state
+ * @param {{
+ *   firstIndex: number,
+ *   firstOffset: number,
+ *   lastIndex: number,
+ * }} columnAnchor
+ * @param {{
+ *   endBufferCol: number,
+ *   endViewportCol: number,
+ *   firstBufferCol: number,
+ *   firstViewportCol: number,
+ * }} columnRange
+ * @private
+ */
+function calculateRenderedColumnGroups(state, columnAnchor, columnRange) {
+  const bufferColumnCount = 0; // TODO (pradeep): calculate this similar to bufferRowCount
+  const { columnGroupBufferSet, columnGroupOffsetIntervalTree } = state;
+
+  const { availableScrollWidth, scrollableColumns, scrollableColumnGroups, columnGroupIndex } = columnWidths(state);
+  const columnCount = scrollableColumns.length;
+  const columnGroupCount = scrollableColumnGroups.length;
+
+  if (columnCount === 0 || columnGroupCount === 0) {
+    state.firstColumnGroupIndex = 0;
+    state.endColumnGroupIndex = 0;
+    state.columnGroupsToRender = [];
+    state.columnGroupOffsets = {};
+    return;
+  }
+
+  // get first and last scrollable columns in the view port
```

Comment:
tricky, will explain below in the selector file that calculates these indices 

* basically we find the first and last scrollable columns in the viewport
* then get their column groups
* which gives us range of column groups inside viewport